crypto/stack/stack.c

```diff
@@ -429,14 +438,32 @@ void OPENSSL_sk_zero(OPENSSL_STACK *st)
 }
 
 void OPENSSL_sk_pop_free(OPENSSL_STACK *st, OPENSSL_sk_freefunc func)
+{
+    if (st == NULL)
+        return;
+
+    OPENSSL_sk_pop_free_thunk(st, st->free_thunk, func);
+}
+
+void OPENSSL_sk_pop_free_thunk(OPENSSL_STACK *st, OPENSSL_sk_freefunc_thunk free_thunk,
```

Comment:
It is not clear to me why do we need both OPENSSL_sk_set_thunks() and OPENSSL_sk_pop_free_thunk() - either we could have OPENSSL_sk_set_thunks() and reimplement OPENSSL_sk_pop_free() in terms of calling the thunk if set or we could just have OPENSSL_sk_pop_free_thunk() with 3 args and then we do not need the OPENSSL_sk_set_thunks() function. Or am I missing something?

Reply:
Fixed. Please have a look.